src/schema/partner_artist.js

```diff
@@ -97,26 +96,10 @@ export const partnersForArtist = (artist_id, options, loader) => {
   }
 
   return loader(gravityArgs).then(({ body, headers }) => {
-    const connection = connectionFromArraySlice(body, options, {
+    return connectionFromArraySlice(body, options, {
       arrayLength: headers["x-total-count"],
       sliceStart: offset,
+      nodeField: "partner",
```

Comment:
So the `nodeField` is really the foreign-key field that points to the child of the (has many) parent node?

Reply:
I guess once you _know_ that the field is easy to understand, but maybe not very self-explanatory without that knowledge. Is that your issue with it as well?

Reply:
Maybe this could be a resolver? E.g.

```js
  connectionFromArraySlice(body, options, {
      arrayLength: headers["x-total-count"],
      sliceStart: offset,
      resolveNode: edge => edge.partner,
    })
```

This would be nice because it also would allow for the case where the node still needs to be fetched at all, e.g.

```js
  connectionFromArraySlice(body, options, {
      arrayLength: headers["x-total-count"],
      sliceStart: offset,
      resolveNode: edge => partnerLoader(edge.partner_id),
    })
```

(Which means that the implement needs to take into account the possibility that `resolveNode` returns a `Promise`.)

Reply:
It's more about the whole concept- basically an option to this API (whether its `nodeField` or something else entirely), which indicates that, essentially, the input data is in fact the edges (and not the nodes). And then additionally, there's a field in there which should _actually_ be treated as the node.

It seems sort of specific to our use case: an API which returns meta/join data (so essentially, the edge), _and_ includes a nested response (in this case, we have nested partners AND nested artists, tho we only care about the nested partner) that can be the node.

So I think my issue isn't this specific naming, but more that I'm not quite sure how to even encapsulate this concept into something concrete, besides just describing our very specific situation.

Reply:
oo I like that `resolveNode` example.

Reply:
The problem with this library has always been that it kind of makes it seem as if edges are supposed to be meaningless, only if you dig deeper do you find that that’s not the case. I would like this library to be more general purpose such that people do start making use of it in the way we are doing, which really isn’t that unique.

I think that the idea of a resolver could work well for all those situations, even when you don’t even use this library to define the edge and node type, but only use it for connection slicing.

Reply:
> (Which means that the implement needs to take into account the possibility that resolveNode returns a Promise.)

That wound up being trivial, since `Promise.resolve` handles all!